pkg/onload/file_writer.go

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: (c) Copyright 2023 Advanced Micro Devices, Inc.
+package onload
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+)
+
+// A tiny helper to write into the Onload sysfs files.
+type FileWriter struct {
+	path string
+	mask fs.FileMode
+}
```

Comment:
I'm not sure that this justifies it existence as a separate struct / file.
There aren't any tests that mock this, so I don't see why there can't be a `writeOnloadParameterFile` function in `kernel_parameters.go`

Reply:
Yep, it's a good idea.